### DISCUSSION TITLE
Option for back button/gesture to open side menu before closing app

### DISCUSSION THREAD

**Discussion by D4RKONION** [spacecowboy/Feeder#179] [Ideas]
Hi there,

Apologies if this was suggested or if it's already in the app. I looked for both and couldn't find it anywhere.

It would be nice for the general user flow of the app for back to open the side menu, and then a second back would close* the app. This is particularly useful for those using gesture navigation rather than the 3 button navigation, as swiping out the side menu from the edge instead quits the app. This would probably be added as an optional setting that was off by default.

You can see an example of this functionality in the Discord app.

Again. Apologies if I've missed this setting or discussion!
Paul 

Edit: close was autocorrected to choose, whoops!

**Comment by nvmkpk** [comment]
Yes, it should be an option that we can enable.

**Comment by gl-dev** [comment]
Oh yes, I couldn't agree more! Please consider implementation of this feature.

**Comment by awqk** [comment]
I think I will never learn that the back button is doing something unexpected, like leaving the App. Please go to the feeds list instead, as suggested. This should be the default action.

**Reply by spacecowboy** [reply]
back button typically exits apps.

personally I want this behavior so that's that

**Comment by gl-dev** [comment]
Maybe implement as a user preference?

**Comment by D4RKONION** [comment]
> back button typically exits apps.
> 
> personally I want this behavior so that's that

Absolutely the back button should exit the app, but it's not unusual for the back button to pull open the sidebar before attempting to exit the app. If user flow is centred around going to the sidebar to navigate "forwards" to feeds then for them they expect the back button to bring them "back" to the sidebar (the home state of the app from this perspective).

Obviously it's your app and I'm not demanding anything here, I'm just saying an option would be appreciated 😊

EDIT: oh, and of course there's the issue that when using gesture navigation the user has to pull from the centre of the screen to pull out the sidebar at the moment; as pulling from the side of the screen closes the app.

**Reply by spacecowboy** [reply]
You are all welcome to submit a PR with this functionality as well as a settings toggle.
